src/components/panel/panel.spec.js

```diff
@@ -105,6 +108,62 @@ describe('$mdPanel', function() {
     expect(PANEL_EL).not.toExist();
   });
 
+  it('should call onDomAdded if provided when adding the panel to the DOM', function() {
+    var onDomAdded = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    panelRef.attachOnly().then(function() {
+      expect(onDomAdded).toHaveBeenCalled();
+      expect(PANEL_EL).toExist();
+    });
+  });
+
+  it('should call onOpenComplete if provided after adding the panel to the DOM and animating', function() {
+    var onOpenComplete = jasmine.createSpy();
+    var animationHookConfig = angular.extend({'onOpenComplete': onOpenComplete}, DEFAULT_CONFIG);
+
+    panelRef = $mdPanel.create(animationHookConfig);
+
+    openPanel();
```

Comment:
These lines can be combined, `openPanel` takes an optional config and if the panelRef isn't created yet, creates one for you. This should be:

`openPanel(animationHookConfig);`